LibWeb: Implemented shadow to canvas2d

uysalibov opened the pull request:
Implemented shadowOffsetX, shadowOffsetY and shadowColor from CanvasShadowStyles.

kalenikaliaksandr reviewed (changes requested):
could you add a [screenshot test](https://github.com/LadybirdBrowser/ladybird/tree/master/Tests/LibWeb/Screenshot) for the change?

uysalibov commented:
I've added screenshot test. @kalenikaliaksandr

kalenikaliaksandr reviewed (changes requested):
Please squash "[LibWeb: Fix lint error on license text](https://github.com/LadybirdBrowser/ladybird/pull/1848/commits/04eda1c5fe1e7b14c48fa51c9ea8065eff9537a3)" into commit it fixes up.

kalenikaliaksandr commented on Userland/Libraries/LibWeb/HTML/CanvasRenderingContext2D.cpp:
```diff
@@ -719,4 +723,95 @@ void CanvasRenderingContext2D::set_global_alpha(float alpha)
+    did_draw(path_to_fill.bounding_box());
+}
+
+void CanvasRenderingContext2D::shadow_to_stroke_internal(Gfx::Path const& path)
```
`shadow_to_stroke_internal` sounds awkward. maybe `paint_shadow_for_stroke_internal`?

kalenikaliaksandr commented on Userland/Libraries/LibWeb/HTML/CanvasRenderingContext2D.cpp:
```diff
@@ -719,4 +723,95 @@ void CanvasRenderingContext2D::set_global_alpha(float alpha)
+        return;
+    }
+}
+void CanvasRenderingContext2D::shadow_to_fill_internal(Gfx::Path const& path, Gfx::WindingRule winding_rule)
```
`shadow_to_fill_internal` is also awkward name

kalenikaliaksandr commented on Userland/Libraries/LibWeb/HTML/CanvasRenderingContext2D.cpp:
```diff
@@ -719,4 +723,95 @@ void CanvasRenderingContext2D::set_global_alpha(float alpha)
+
+    auto& state = this->drawing_state();
+
+    painter->save(); // Save the current state of the painter
```
having comments for lines that call save() or restore() is a noise that's not helpful while reading code. please remove them.

uysalibov commented:
Hey @kalenikaliaksandr thanks for review. This is my first PR. I'm learning too many stuff. 

- Squashed "Error lint" commit
- Removed meaningless comments
- Changed shadow internal function names to which is you want.

kalenikaliaksandr reviewed (changes requested):
could you change commit name to "LibWeb: Implemented shadows support for CanvasRenderingContext2D".

"LibWeb: Implemented shadow to canvas2d" sounds strange.
